scoper.inc.php

```diff
@@ -0,0 +1,226 @@
+			->ignoreVCS( true )
+			->ignoreDotFiles( true )
+			->name( '*.php' )
+			->notName(
+				[
+					'amp.php',
+					'amp-frontend-actions.php',
+					'amp-helper-functions.php',
+					'amp-post-template-functions.php',
+					'class-amp-autoloader.php',
+					'class-amp-comment-walker.php',
+					'class-amp-http.php',
+					'class-amp-post-type-support.php',
+					'class-amp-service-worker.php',
+					'class-amp-theme-support.php',
+					'deprecated.php',
+					'amp-enabled-classic-editor-toggle.php',
+					'amp-paired-browsing.php',
+					'class-amp-content.php',
+					'class-amp-post-template.php',
+					'reader-template-loader.php',
+				]
+			)
+			->exclude(
+				[
+					'assets',
+					'back-compat',
+					'bin',
+					'docs',
+					'includes/admin',
+					'includes/cli',
+					'includes/embeds',
+					'includes/options',
+					'includes/settings',
+					'includes/validation',
+					'includes/widgets',
+					'patches',
+					'src',
+					'tests',
+					'vendor',
+					'wp-assets',
+					'tests',
+				]
+			)
+			->in(
+				[
+					'vendor/ampproject/amp-wp/includes',
+					'vendor/ampproject/amp-wp/src/RemoteRequest',
+					'vendor/ampproject/amp-wp/templates',
+				]
+			)
+			->append( [ 'vendor/ampproject/amp-wp/composer.json' ] ),
+
+		// AMP Common + Optimizer.
+		Finder::create()
+			->files()
+			->ignoreVCS( true )
+			->ignoreDotFiles( true )
+			->name( '*.php' )
+			->in(
+				[
+					'vendor/ampproject/amp-wp/lib',
+				]
+			),
+
+		// FasterImage (used by AMP_Img_Sanitizer).
+		Finder::create()
+			->files()
+			->ignoreVCS( true )
+			->ignoreDotFiles( true )
+			->name( '*.php' )
+			->exclude(
+				[
+					'tests',
+				]
+			)
+			->in( 'vendor/fasterimage/fasterimage' )
+			->append( [ 'vendor/fasterimage/fasterimage/composer.json' ] ),
+
+		// PHP-CSS-Parser (used by AMP_Style_Sanitizer).
+		Finder::create()
+			->files()
+			->ignoreVCS( true )
+			->ignoreDotFiles( true )
+			->name( '*.php' )
+			->exclude(
+				[
+					'tests',
+				]
+			)
+			->in( 'vendor/sabberworm/php-css-parser' )
+			->append( [ 'vendor/sabberworm/php-css-parser/composer.json' ] ),
+
+		// Main composer.json file so that we can build a classmap.
+		Finder::create()
+			->append( [ 'composer.json' ] ),
+	],
+
+	// Whitelists a list of files. Unlike the other whitelist related features, this one is about completely leaving
+	// a file untouched.
+	// Paths are relative to the configuration file unless if they are already absolute.
+	'files-whitelist'            => [],
+
+	// When scoping PHP files, there will be scenarios where some of the code being scoped indirectly references the
+	// original namespace. These will include, for example, strings or string manipulations. PHP-Scoper has limited
+	// support for prefixing such strings. To circumvent that, you can define patchers to manipulate the file to your
+	// heart contents.
+	//
+	// For more see: https://github.com/humbug/php-scoper#patchers.
+	'patchers'                   => [
```

Comment:
OMG. Now I feel lucky Site Kit doesn't use WordPress plugins as dependency - class-based libraries work way better with this than global WordPress function mess 😆

Reply:
😅 yes you're lucky!

I was hoping the [whitelist options](https://github.com/humbug/php-scoper#whitelist) were a bit better. Maybe I need to give them another try though.